tests/integration/components/async-button-test.js

```diff
@@ -0,0 +1,63 @@
+import Ember from 'ember';
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+
+moduleForComponent('async-button', 'Integration | Component | async button', {
+  integration: true
+});
+
+test('it responds to a fulfilled closure promise', function(assert) {
+  let done = assert.async();
+
+  assert.expect(2);
+
+  this.set('default', 'Save');
+  this.set('pending', 'Saving');
+  this.set('fulfilled', 'Saved');
+
+  this.on('closurePromise', function() {
+    return new Ember.RSVP.Promise((fulfill) => {
+      Ember.run.later(() => {
```

Comment:
Lets use `Ember.run.next` to ensure async, no need to force a 500ms delay on the test suite...